packages/cmd/src/syntax/SyntaxValidator.ts

```diff
@@ -294,8 +294,20 @@ export class SyntaxValidator {
 
                 // check if the value of the option conforms to the allowableValues (if any)
                 if (!isNullOrUndefined(optionDef.allowableValues)) {
-                    if (!this.checkIfAllowable(optionDef.allowableValues, commandArguments[optionName])) {
-                        this.invalidOptionError(optionDef, responseObject, commandArguments[optionName]);
+                    // Make a copy of optionDef, so that modifications below are only used in this place
+                    const optionDefCopy: ICommandOptionDefinition = JSON.parse(JSON.stringify(optionDef));
+                    // Use modified regular expressions for allowable values to check and to generate error
+                    optionDefCopy.allowableValues.values = optionDef.allowableValues.values.map((regex) => {
+                        // If user-defined regex already contains ^ and/or $,
+                        // we assume user is aware of the usage and thus keep user's definition.
+                        if (regex.startsWith("^") || regex.endsWith("$")) {
```

Comment:
Not sure if this assumption is safe. It may lead to unexpected behavior if users want an exact match for strings containing these characters. For example, if there is a CLI option for currency symbols, `\$` could be an allowable value regex that should not be treated as special.

Reply:
Yes you are right. The check `regex.endsWith("$")` is not correct. It should be `regex.endsWith("$") && !regex.endsWith("\\$")`.

But as you said in the last comment that we will treat this as a bugfix, and thus non-breaking change, I will always add ^ and $ to match exactly.